code/modules/chemistry/Reagents-ExplosiveFire.dm

```diff
@@ -720,10 +720,17 @@ datum
 				return 1
 
 			on_mob_life(var/mob/M, var/mult = 1)
-
-				var/mob/living/L = M
-				if(istype(L) && L.getStatusDuration("burning"))
-					L.changeStatus("burning", 2 SECONDS * mult)
+				if (!M)
+					M = holder.my_atom
+				if(istype(M, /mob/living/) && M.getStatusDuration("burning"))
+					M.changeStatus("burning", 2 SECONDS * mult)
+				if(holder.has_reagent("welding fuel"))
+					holder.remove_reagent("welding fuel", 1 * mult)
+				if((M.health > 20) && (probmult(33)))
+					M.take_toxin_damage(1 * mult)
```

Comment:
Sanity check on combining a `probmult` with a "scale damage by `mult`"? I *think* it should be one or the other, but not used `probmult` extensively. Second opinion from someone else?

Reply:
multiplying values is preferable over probmult, and both at once should not be used